src/main.rs

```diff
@@ -116,14 +121,16 @@ fn main() {
         });
     }
 
-    let sleep_duration = std::time::Duration::new(config.run_interval, 0);
     let bitbucket = bitbucket::Bitbucket::new(&config.bitbucket, &fanout);
     if let Some(t) = config.clone().telegram {
         if t.enabled {
             t.announce_from(fanout.subscribe()).expect("Failed to authenticate with Telegram");
         }
     }
 
+    let schedule: CronSchedule = CronSchedule::parse(config.cron_interval).unwrap();
```

Comment:
It's not a fixed interval though, from the cron. The general idea of my code, is that we use the cron pattern given in the config to get the next run time. By deducting the next run time, and now, we get a `Duration` that defines the amount of time from now till the next run time.